### ISSUE TITLE
The setting for changing the user-agent doesn't work, you have to clobber the function

### ISSUE THREAD

**Issue by avar** [djcb/mu#703]
I found a bug which I want to fix, but I'm not quite sure how I should
fix it since there's some idiom at play I don't grok.

I wanted to override the User-Agent it sends (to mention my linux
distro) and I found myself having to do this:
https://github.com/avar/dotemacs/blob/master/.emacs#L518

The reason is that the docs here and the implementation don't match:
https://github.com/djcb/mu/blob/master/mu4e/mu4e-draft.el#L80

I.e. it pays no attention to mu4e-user-agent, but says it will.

Now, I thought it should be trivial to just make mu4e-user-agent
customizable and have an (if) in that function

But I don't understand what this whole thing is about:
https://github.com/djcb/mu/blob/master/mu4e/mu4e-compose.el#L563

@tarsius submitted a fix for you to add a shimmy layer for this in
2012:
https://github.com/djcb/mu/commit/6424a82d60183c712b9d821e4077d284307d8fb5

AFAICT this whole shimmy layer isn't needed and isn't used.

So I'm wondering if I should just make the function actually use
mu4e-user-agent, or add some new mu4e-draft-* variable to set the UA

I haven't used the interactive customize that much, and this seems to
be a shimmy for that, which may explain my confusion.

So basically. I *think* the fix is to rm that stuff added in
https://github.com/djcb/mu/commit/6424a82d60183c712b9d821e4077d284307d8fb5
and instead add it as a customizable variable that's used rom
mu4e-draft.el, or since it never actually worked maybe put it in the
mu4e-draft namespace.

So I can submit a patch for this. But wanted some feedback first since
it involves an elisp idiom I'm not familiar with.

**Comment by djcb** [comment]
I have added a variable `mu4e-user-agent-string` that you can customize - this is about the User-Agent header.

Note, the second meaning of the term 'user-agent' (as in `define-mail-user-agent`) is something orthogonal to that.
